hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieClusteringConfig.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.config;
+
+import org.apache.hudi.common.config.DefaultHoodieConfig;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * Clustering specific configs.
+ */
+public class HoodieClusteringConfig extends DefaultHoodieConfig {
+
+  public static final String SCHEDULE_CLUSTERING_STRATEGY_CLASS = "hoodie.clustering.schedule.strategy.class";
+  public static final String DEFAULT_SCHEDULE_CLUSTERING_STRATEGY_CLASS =
```

Comment:
Looks like the execution framework details are in the configs. Would be nice to have a wrapper around reflection utils that can pick the right execution engine based class but keep the strategy class name generic...not sure if it's too late for that..

Reply:
This is just the class name. It is possible to support both flink and spark clients with same class. Initial implementation is primarily targeted for Spark. So i explicitly called the class Spark.